tests/itests-cxf/src/test/java/org/apache/camel/kafkaconnector/cxf/sink/GreeterImpl.java

```diff
@@ -15,16 +15,16 @@
  * limitations under the License.
  */
 
-package org.apache.camel.kafkaconnector.common.test;
+package org.apache.camel.kafkaconnector.cxf.sink;
 
-import java.util.Map;
-import java.util.concurrent.ExecutionException;
+import java.util.logging.Logger;
 
-/**
- * A producer of test messages
- */
-public interface TestMessageProducer<T> {
-    Map<String, String> messageHeaders(T text, int current);
-    T testMessageContent(int current);
-    void produceMessages() throws ExecutionException, InterruptedException;
+
+public class GreeterImpl extends org.apache.hello_world_soap_http.GreeterImpl {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Class names shouldn’t shadow simple name of superclass (NM_SAME_SIMPLE_NAME_AS_SUPERCLASS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-class-names-shouldn-t-shadow-simple-name-of-superclass-nm-same-simple-name-as-superclass).
This class has a simple name that is identical to that of its superclass, except that its superclass is in a different package (e.g., alpha.Foo extends beta.Foo). This can be exceptionally confusing, create lots of situations in which you have to look at import statements to resolve references and creates many opportunities to accidentally define methods that do not override methods in their superclasses.

Reply:
Thanks, but this is how cxf services are usually generated.